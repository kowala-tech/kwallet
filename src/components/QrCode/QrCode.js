import React from "react";
import PropTypes from "prop-types";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
// Component Related Imports
import styles from "./styles";
import { QRCode } from "react-qr-svg";

class QrCode extends React.Component {

	render() {
		const {
			classes,
			text,
			...rest
		} = this.props;

		return (
			<div className={classes.root}>
				<QRCode
					value={text}
					{...rest}
				/>
			</div>
		);
	}
}

QrCode.propTypes = {
	classes: PropTypes.object.isRequired,
	text: PropTypes.string.isRequired
};

export default withStyles(styles)(QrCode);
