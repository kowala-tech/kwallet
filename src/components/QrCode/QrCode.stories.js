import React from "react";
import { storiesOf } from "@storybook/react";
// Story related component imports
import QrCode from "./QrCode";

storiesOf("QrCode", module)
	.add("default", () =>
		(<QrCode
			text="this is text"
		/>)
	);
