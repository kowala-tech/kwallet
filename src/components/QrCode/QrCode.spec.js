import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import QrCode from "./QrCode";

describe("QrCode", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<QrCode
					text="this is text"
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
