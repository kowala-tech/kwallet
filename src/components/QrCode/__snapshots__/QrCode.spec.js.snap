// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrCode it should match the snapshot 1`] = `
<QrCode
  classes={Object {}}
  text="this is text"
/>
`;
